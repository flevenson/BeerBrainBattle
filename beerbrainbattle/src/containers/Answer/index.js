import React from 'react';
import './Answer.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';

export const Answer = (props) => {
  if(!Object.keys(props.question).length){
    props.history.push('/')
  } 

  let correctAnswer

  if(Object.keys(props.question).length){
    correctAnswer = props.question.answers.find(answer => answer.correct === true)
  return(
    <div>
      <p className='correct-answer-title'>The Correct Answer Is</p>
      <p>{correctAnswer.answer}</p>
    </div>
    )
  } else {
    return (
      <div>
      </div>
    )
  }

}

export const mapStateToProps = (state) => ({
  players: state.players,
  question: state.question
})

export default withRouter(connect(mapStateToProps)(Answer))