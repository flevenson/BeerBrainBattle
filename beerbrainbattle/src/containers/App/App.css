header {
  display: flex;
  align-items: center;
  height: 150px;
  justify-content: center;
  font-size: 4rem;
  font-family: 'Lily Script One', cursive;
}

.heading-beer {
  color: #2E2453;
}

.heading-brain {
  color: #C7EAF0;
}

.heading-battle {
  color: #F8BFA8;
}

main {
  background-image: url('../../assets/beer-can.svg');
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
  height: 720px;
  display: flex;
  align-items: center;
  justify-content: center;
}