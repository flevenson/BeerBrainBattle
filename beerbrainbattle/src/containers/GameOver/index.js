import React from 'react';
import './GameOver.css';
import Proptypes from 'prop-types';
import { connect } from 'react-redux';
import { addPlayers } from '../../actions';
import { withRouter } from 'react-router';
import * as BeerData from '../../assets/BeerData.js';

export const GameOver = (props) => {

  if(!Object.keys(props.question).length){
    props.history.push('/')
  } 

  let correctAnswer

  if(Object.keys(props.question).length){
    correctAnswer = props.question.answers.find(answer => answer.correct === true)
  return(
    <div className='answer-container'>
      <div className='answer'>
        <h1 className='game-over'>GAME OVER</h1>
        <p className='correct-answer-title'>The Correct Answer Was:</p>
        <p className='correct-answer-answer'>{correctAnswer.answer}</p>
        <p className='award'>You Won {props.prize}</p>
      </div>
      <button 
        onClick={() => {
          props.addPlayers(0)
          props.history.push('/')
        }} 
        className='next-round'>Play Again!</button>
    </div>
    )
  } else {
    return (
      <div>
      </div>
    )
  }


}

export const mapStateToProps = (state) => ({
  players: state.players,
  question: state.question,
  prize: state.prize
})

export const mapDispatchToProps = (dispatch) => ({
  addPlayers: (players) => dispatch(addPlayers(players)),
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(GameOver))