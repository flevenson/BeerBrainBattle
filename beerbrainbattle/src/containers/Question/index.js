import React from 'react';
import './Question.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { addVote } from '../../actions'

export const Question = (props) => {

  if(!Object.keys(props.question).length && !props.players){
    props.history.push('/')
  } 
  
  if(Object.keys(props.question).length){

  const answers = props.question.answers.map(answer => 
    <button 
      className='answer-button'
      onClick={() => props.addVote(answer.answer)}
      >{answer.answer}   {answer.numVotes}
    </button>
)

  const totalVotes = props.question.answers.reduce((allVotes, answer) => {
    allVotes += answer.numVotes
    return allVotes
  }, 0)

  console.log(totalVotes)

  if(totalVotes === parseInt(props.players) && Object.keys(props.question).length) {
    props.history.push('/answer')
  }


    return (
      <div className='question-holder'>
        <div className='question'>
          <h1>{props.question.question}</h1>
        </div>
        <div className='answers'>
          { answers }
        </div>
      </div>
      )
  }
  else {
    return (
      <div>
      </div>
      )
  }

}

export const mapStateToProps = (state) => ({
  players: state.players,
  question: state.question
})

export const mapDispatchToProps = (dispatch) => ({
  addVote: (answer) => dispatch(addVote(answer)),
})


export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Question))