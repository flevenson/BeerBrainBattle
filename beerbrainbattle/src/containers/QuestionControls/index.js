import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addPlayers, addQuestion, addPrize } from '../../actions'
import './QuestionControls.css';
import PropTypes from 'prop-types';
import * as BeerData from '../../assets/BeerData.js';
import * as API from '../../utils'
import { withRouter } from 'react-router';

export class QuestionControls extends Component {
  constructor(props) {
    super(props)
    this.state = {
      category: '',
      difficulty: '',
      numPlayers: '',
      prize: '',
      showCategories: false,
      showDifficulty: false,
    }
  }


  toggleShowCategories = () => {
    this.setState({
      showCategories: !this.state.showCategories
    })
  }

  setCategory = (event) => {
    const category = event.target.innerText

    this.setState({
      category,
      showCategories: !this.state.showCategories
    })
  }

  toggleShowDifficulty = () => {
    this.setState({
      showDifficulty: !this.state.showDifficulty
    })
  }

  setDifficulty = (event) => {
    const difficulty = event.target.innerText

    this.setState({
      difficulty,
      showDifficulty: !this.state.showDifficulty
    })
  }  

  handleInputChange = (event) => {
    const { name, value } = event.target
    this.setState({
      [name]: value
    })
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { players, category, difficulty, numPlayers, prize } = this.state
    if(this.props.players !== 0){
      this.setState({
        numPlayers: this.props.players
      })
    }
    const question = await API.fetchRandomQuestion(category, difficulty);
    this.props.addQuestion(question);
    this.props.addPlayers(numPlayers);
    this.makePrize();
    await this.props.history.push('/question')  
}

  makePrize = () => {
    if(!this.state.prize.length) {
      const randomBeer = BeerData.beerData[Math.floor(Math.random() * 100)]
      const randomPrize = `${randomBeer.year} ${randomBeer.beerName} from  ${randomBeer.brewery} worth $${randomBeer.value}`
      this.props.addPrize(randomPrize)
    } else {
      this.props.addPrize(this.state.prize)
    }
  }

  render() {

  const { showDifficulty, showCategories, category, difficulty, numPlayers, prize } = this.state;


  const categoryOptions = BeerData.categories.map(categoryData => 
    <li 
      className={ 
        showCategories 
        ? 'dropdown-option' 
        : 'hidden' } 
      onClick={ this.setCategory }>{ categoryData } 
    </li>
  )

    return(
      <form className='question-controls'>
        <div>
          <div className='dropdown-title' onClick={ this.toggleShowCategories}>{ category.length ? category : 'Category' }</div>
          <ul className='category-holder'>
            { categoryOptions }
          </ul>
        </div>
        <div>
          <div className='dropdown-title' onClick={ this.toggleShowDifficulty}>{ difficulty.length ? difficulty : 'Difficulty'}</div>
          <ul>
            <li 
              className={ 
                showDifficulty 
                ? 'dropdown-option' 
                : 'hidden'} 
              onClick={this.setDifficulty}>Easy
            </li>
            <li 
              className={ 
                showDifficulty 
                ? 'dropdown-option' 
                : 'hidden'} 
              onClick={this.setDifficulty}>Medium
            </li>
            <li 
              className={ 
                showDifficulty 
                ? 'dropdown-option' 
                : 'hidden'} 
              onClick={this.setDifficulty}>Hard
            </li>
          </ul>
        </div>
        <input 
          className={this.props.players ? 'hidden' : 'dropdown-title'} 
          placeholder='Number Of Players' 
          value={ numPlayers } 
          onChange={this.handleInputChange} 
          name='numPlayers'>
        </input>
        <input 
          className={this.props.prize ? 'hidden' : 'dropdown-title'} 
          placeholder='What will you Wager?' 
          value={ prize } 
          onChange={this.handleInputChange}
          name='prize'>
        </input>
        <button className='dropdown-title' onClick={this.handleSubmit}> Battle </button>
      </form>
    )
  }
}

export const mapStateToProps = (state) => ({
  players: state.players
})

export const mapDispatchToProps = (dispatch) => ({
  addPlayers: (players) => dispatch(addPlayers(players)),
  addQuestion: (question) => dispatch(addQuestion(question)),
  addPrize: (prize) => dispatch(addPrize(prize))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(QuestionControls))