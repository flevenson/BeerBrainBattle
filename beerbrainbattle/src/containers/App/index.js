import React, { Component } from 'react';
import logo from '../../logo.svg';
import './App.css';
import PropTypes from 'prop-types';
import { Route, withRouter } from 'react-router-dom';
import QuestionControls from '../QuestionControls';
import Question from '../Question'
import Answer from '../Answer'
import { connect } from 'react-redux';

export const App = (props) => {

    return (
      <div className="App">
        <header>
          <h1 className='app-heading'>
            <span className='heading-beer'>Beer</span>
            <span className='heading-brain'>Brain</span>
            <span className='heading-battle'>Battle</span>
          </h1>
            <span className='player-count-holder'>
                <h1 className='player-count'>{props.players}</h1>
                <h1 className='players'>players</h1>
            </span>
        </header>
        <main className='app-main'>
          <Route 
            exact path='/'
            render={() => <QuestionControls />}
          />
          <Route 
            exact path='/question'
            render={() => <Question />}
          />
          <Route 
            exact path='/answer'
            render={() => <Answer />}
          />
        </main>
      </div>
    )

}

export const mapStateToProps = (state) => ({
  players: state.players
})

// export const mapDispatchToProps = (dispatch) => ({

// })

export default withRouter(connect(mapStateToProps)(App));
