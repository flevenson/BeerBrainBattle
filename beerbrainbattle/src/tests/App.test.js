import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from '../containers/App';
import { mockPrize, mockPlayers } from './testMocks.js'

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App 
      prize={ mockPrize }/>);
  })

  describe('App Component', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should parse the players from state', () => {
      const mockState = {
        players: mockPlayers,
        prize: mockPrize
      }
      const expected = 4

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps.players).toEqual(expected)
    })

    it('should parse the prize from state', () => {
      const mockState = {
        players: mockPlayers,
        prize: mockPrize
      }

      const expected = mockPrize

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps.prize).toEqual(expected)
    })
  })
})