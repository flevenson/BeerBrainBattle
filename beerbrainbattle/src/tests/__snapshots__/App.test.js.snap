// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    prize="1 Whole Beer"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1
            className="app-heading"
          >
            <span
              className="heading-beer"
            >
              Beer
            </span>
            <span
              className="heading-brain"
            >
              Brain
            </span>
            <span
              className="heading-battle"
            >
              Battle
            </span>
          </h1>
          <span
            className="hidden"
          >
            <h1
              className="player-count"
            />
            <h1
              className="players"
            >
              players
            </h1>
          </span>
        </header>,
        <main
          className="app-main"
        >
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/question"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/answer"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/gameOver"
              render={[Function]}
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </main>,
        <footer>
          <h1
            className="prize-text"
          >
            Current Prize: 
            1 Whole Beer
          </h1>
        </footer>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="app-heading"
            >
              <span
                className="heading-beer"
              >
                Beer
              </span>
              <span
                className="heading-brain"
              >
                Brain
              </span>
              <span
                className="heading-battle"
              >
                Battle
              </span>
            </h1>,
            <span
              className="hidden"
            >
              <h1
                className="player-count"
              />
              <h1
                className="players"
              >
                players
              </h1>
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="heading-beer"
                >
                  Beer
                </span>,
                <span
                  className="heading-brain"
                >
                  Brain
                </span>,
                <span
                  className="heading-battle"
                >
                  Battle
                </span>,
              ],
              "className": "app-heading",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Beer",
                  "className": "heading-beer",
                },
                "ref": null,
                "rendered": "Beer",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Brain",
                  "className": "heading-brain",
                },
                "ref": null,
                "rendered": "Brain",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Battle",
                  "className": "heading-battle",
                },
                "ref": null,
                "rendered": "Battle",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="player-count"
                />,
                <h1
                  className="players"
                >
                  players
                </h1>,
              ],
              "className": "hidden",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "player-count",
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "players",
                  "className": "players",
                },
                "ref": null,
                "rendered": "players",
                "type": "h1",
              },
            ],
            "type": "span",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/question"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/answer"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/gameOver"
              render={[Function]}
            />
            <Route
              component={[Function]}
            />
          </Switch>,
          "className": "app-main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/question"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/answer"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/gameOver"
                render={[Function]}
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/question",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/answer",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/gameOver",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="prize-text"
          >
            Current Prize: 
            1 Whole Beer
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Current Prize: ",
              "1 Whole Beer",
            ],
            "className": "prize-text",
          },
          "ref": null,
          "rendered": Array [
            "Current Prize: ",
            "1 Whole Beer",
          ],
          "type": "h1",
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1
              className="app-heading"
            >
              <span
                className="heading-beer"
              >
                Beer
              </span>
              <span
                className="heading-brain"
              >
                Brain
              </span>
              <span
                className="heading-battle"
              >
                Battle
              </span>
            </h1>
            <span
              className="hidden"
            >
              <h1
                className="player-count"
              />
              <h1
                className="players"
              >
                players
              </h1>
            </span>
          </header>,
          <main
            className="app-main"
          >
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/question"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/answer"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/gameOver"
                render={[Function]}
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </main>,
          <footer>
            <h1
              className="prize-text"
            >
              Current Prize: 
              1 Whole Beer
            </h1>
          </footer>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="app-heading"
              >
                <span
                  className="heading-beer"
                >
                  Beer
                </span>
                <span
                  className="heading-brain"
                >
                  Brain
                </span>
                <span
                  className="heading-battle"
                >
                  Battle
                </span>
              </h1>,
              <span
                className="hidden"
              >
                <h1
                  className="player-count"
                />
                <h1
                  className="players"
                >
                  players
                </h1>
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="heading-beer"
                  >
                    Beer
                  </span>,
                  <span
                    className="heading-brain"
                  >
                    Brain
                  </span>,
                  <span
                    className="heading-battle"
                  >
                    Battle
                  </span>,
                ],
                "className": "app-heading",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Beer",
                    "className": "heading-beer",
                  },
                  "ref": null,
                  "rendered": "Beer",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Brain",
                    "className": "heading-brain",
                  },
                  "ref": null,
                  "rendered": "Brain",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Battle",
                    "className": "heading-battle",
                  },
                  "ref": null,
                  "rendered": "Battle",
                  "type": "span",
                },
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="player-count"
                  />,
                  <h1
                    className="players"
                  >
                    players
                  </h1>,
                ],
                "className": "hidden",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "player-count",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "players",
                    "className": "players",
                  },
                  "ref": null,
                  "rendered": "players",
                  "type": "h1",
                },
              ],
              "type": "span",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/question"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/answer"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/gameOver"
                render={[Function]}
              />
              <Route
                component={[Function]}
              />
            </Switch>,
            "className": "app-main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  exact={true}
                  path="/question"
                  render={[Function]}
                />,
                <Route
                  exact={true}
                  path="/answer"
                  render={[Function]}
                />,
                <Route
                  exact={true}
                  path="/gameOver"
                  render={[Function]}
                />,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/question",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/answer",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/gameOver",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="prize-text"
            >
              Current Prize: 
              1 Whole Beer
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Current Prize: ",
                "1 Whole Beer",
              ],
              "className": "prize-text",
            },
            "ref": null,
            "rendered": Array [
              "Current Prize: ",
              "1 Whole Beer",
            ],
            "type": "h1",
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
